optuna/integration/allennlp.py

```diff
@@ -141,7 +115,9 @@ def _get_environment_variables_for_pruner() -> Dict[str, Optional[Union[str, int
     kwargs = {}
     for key in keys.split(","):
         key_without_prefix = key.replace("{}_".format(_PREFIX), "")
-        kwargs[key_without_prefix] = _infer_and_cast(os.getenv(key))
+        value = os.getenv(key)
+        assert value is not None
+        kwargs[key_without_prefix] = eval(value)
```

Comment:
Of course, this implementation could be problematic when some new pruner having a parameter that is hard to restore using `eval`. However, the problems would only occur in this integration and we'll be exploring other solutions when we are in need.